src/main/java/org/prebid/server/cookie/CookieSyncService.java

```diff
@@ -0,0 +1,473 @@
+
+        final UsersyncMethod usersyncMethod = biddersContext.bidderUsersyncMethod().get(bidder);
+        final Privacy privacy = cookieSyncContext.getPrivacyContext().getPrivacy();
+        final String hostCookieUid = uidsCookieService.hostCookieUidToSync(routingContext, cookieFamilyName);
+
+        final UsersyncInfo usersyncInfo = toUsersyncInfo(usersyncMethod, cookieFamilyName, hostCookieUid, privacy);
+
+        return BidderUsersyncStatus.builder()
+                .bidder(cookieFamilyName) // we are syncing cookie-family-names instead of bidder codes
+                .noCookie(true)
+                .usersync(usersyncInfo)
+                .build();
+    }
+
+    private List<BidderUsersyncStatus> errorStatuses(CookieSyncContext cookieSyncContext) {
+        if (!cookieSyncContext.isDebug()) {
+            return Collections.emptyList();
+        }
+
+        final BiddersContext biddersContext = cookieSyncContext.getBiddersContext();
+        return biddersContext.rejectedBidders().entrySet().stream()
+                .map(bidderWithReason ->
+                        errorStatus(bidderWithReason.getKey(), bidderWithReason.getValue(), biddersContext))
+                .filter(BidderUsersyncStatus::isError)
+                .toList();
+    }
+
+    private BidderUsersyncStatus errorStatus(String bidder, RejectionReason reason, BiddersContext biddersContext) {
+        final String cookieFamilyName = bidderCatalog.cookieFamilyName(bidder).orElse(bidder);
+        BidderUsersyncStatus.BidderUsersyncStatusBuilder builder = BidderUsersyncStatus.builder()
+                .bidder(cookieFamilyName);
+
+        final boolean requested = biddersContext.isRequested(bidder);
+        final boolean coopSync = biddersContext.isCoopSync(bidder);
+
+        builder = switch (reason) {
+            case INVALID_BIDDER -> builder.conditionalError(requested, "Unsupported bidder");
+            case DISABLED_BIDDER -> builder.conditionalError(requested, "Disabled bidder");
+            case REJECTED_BY_TCF -> builder.conditionalError(requested || coopSync, "Rejected by TCF");
+            case REJECTED_BY_CCPA -> builder.conditionalError(requested || coopSync, "Rejected by CCPA");
+            case UNCONFIGURED_USERSYNC -> builder.conditionalError(requested, "No sync config");
+            case REJECTED_BY_FILTER -> builder.conditionalError(requested || coopSync, "Rejected by request filter");
+            case ALREADY_IN_SYNC -> builder.conditionalError(requested, "Already in sync");
+        };
+
+        return builder.build();
+    }
+
+    private List<BidderUsersyncStatus> warningStatuses(Set<String> biddersToSync, CookieSyncContext cookieSyncContext) {
+        if (!cookieSyncContext.isDebug()) {
+            return Collections.emptyList();
+        }
+
+        return biddersToSync.stream()
+                .filter(bidder -> cookieSyncContext.getBiddersContext().allowedRequestedBidders().contains(bidder))
```

Comment:
`cookieSyncContext.getBiddersContext().allowedRequestedBidders()::contains`

Reply:
Thought about this approach, and I just could not decide whether to use it or not. Because I dont like that I cant do same thing for negation (like `.filter(bidder -> !rejectedByCcpaBidders.contains(bidder))`) and it adds little inconsistency inside the code. What do you think, @SerhiiNahornyi?

Reply:
Agree with Alex